tests/autoyast/verify_lvm_partitions.pm

```diff
@@ -0,0 +1,29 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+#
+# Summary: Verify lvm partitions after autoyast installation
+# Maintainer: qa-sle-yast <qa-sle-yast@suse.com>
+
+use strict;
+use warnings;
+use parent 'installbasetest';
+use testapi;
+use scheduler 'get_test_suite_data';
+
+sub run {
+    my $test_data = get_test_suite_data();
+
+    my $vg = $test_data->{profile}->{partitioning}->[1]->{drive}->{partitions}->[0]->{partition}->{lvm_group};
+    assert_script_run("vgdisplay $vg");
+    foreach my $partition (@{$test_data->{profile}->{partitioning}->[0]->{drive}->{partitions}}) {
+        assert_script_run("lvdisplay /dev/$vg/$partition->{partition}->{lv_name}");
```

Comment:
Just an idea here, what I think I would do is try to parse each partition and if lvm then validate. This way the module could be more generically used.

Reply:
I don't think this can work, as the test data may strongly vary from one test suite to the other.

Reply:
I agreed, it is too specific, in general we can share test_data for different modules, but the validation of the cloned profile is very specific, we could have included outside somehow (but that is the current design), once you use it for your specific case, then we cannot plug&play module anywhere else. It is better to have separated test data for what you want to validate (also we will save arrows :) )

Reply:
I don't know what is better, adding more test data spares some arrows and may be more readable, but it's redundant, we'd simpify on one side but complicate on the other: a lot of test data. For the other point, we still face the same issue, that the test data for partitioning is not standardized, each test suite uses a different format. It would be quite a hassle at that point to standardize all the test data.

Reply:
Here what is confusing for me, is that `$vg` variable seems to be set using array item with index `1`: 
```
my $vg = $test_data->{profile}->{partitioning}->[1]...
```

But in `foreach` loop we are iterating over the array item with index `0`:
```
foreach my $partition (@{$test_data->{profile}->{partitioning}->[0]...
```

If you just need a value of `lvm_group`, then maybe it makes sense to assign it to separate variable in a test_data and use it without this long chain? Could you please correct me if I'm wrong and you had some idea underneath.

Reply:
Here we have a VG sitting on two partitions (so-called physical volumes, but in this case really just primary partitions) which makes it a bit confusing. $test_data->{profile}->{partitioning}->[0] is the first partition, $test_data->{profile}->{partitioning}->[1] is the second one. 
So if we want to avoid this form, we have just two options : change the original autoyast profile, or add more test data, just for this module. I'd go for the second option as I think it' good to test lvm in various configurations.

Reply:
Also, to clarify, here we iterate over the lvm volumes within the physical partition 0.

Reply:
Anyway, I'll just add more test data :)